src/mapgen_valleys.cpp

```diff
@@ -897,101 +783,150 @@ void MapgenValleys::dustTopNodes()
 }
 
 
-void MapgenValleys::generateSimpleCaves(s16 max_stone_y)
+void MapgenValleys::generateCaves(s16 max_stone_y)
 {
+	if (max_stone_y < node_min.Y)
+		return;
+
 	PseudoRandom ps(blockseed + 72202);
```

Comment:
Unless reverse compatibility is a concern, you should use PcgRandom() since it produces superior randomness.